corehq/apps/domain/auth.py

```diff
@@ -215,7 +216,18 @@ def authenticate(self, request, username, password):
             return None
 
         try:
-            user = User.objects.get(username=username, api_keys__key=password)
+            is_unexpired_filter = (
+                Q(api_keys__expiration_date__isnull=True) | Q(api_keys__expiration_date__gte=datetime.now())
```

Comment:
I think this should use `datetime.utcnow()`.
```suggestion
                Q(api_keys__expiration_date__isnull=True) | Q(api_keys__expiration_date__gte=datetime.utcnow())
```
...or whatever the appropriate alternative is these days, I should know what it is, been seeing warnings in my IDE about `datetime.utcnow` for a while.

Reply:
Not thrilled with this, but I've made the change. `utcnow` is apparently deprecated, and the recommendation to use `datetime.now(datetime.UTC)` relies on the UTC constant which is only supplied in Python 3.11 and onwards. Anyhow, made the change for both methods to use `datetime.now(timezone.utc)` instead.